js/util/string_number_mapping.js

```diff
@@ -0,0 +1,13 @@
+'use strict';
+
+module.exports = StringNumberMapping;
+
+function StringNumberMapping(strings) {
+    this.stringToNumber = {};
+    this.numberToString = [];
+    for (var i = 0; i < strings.length; i++) {
+        var string = strings[i];
+        this.stringToNumber[string] = i;
+        this.numberToString[i] = string;
+    }
+}
```

Comment:
I would be more confused with `DictionaryCoder` if I saw it in the code without seeing this discussion. Can't think of a better name than `StringNumberMapping`, but it definitely needs a clarifying comment / jsdoc.